.github/workflows/build.yml

```diff
@@ -0,0 +1,165 @@
+name: Build
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: 
+      - main
+jobs:
+  versionning:
+    name: Versionning
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.genver.outputs.version }}
+      release: ${{ steps.genver.outputs.release }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Generate Version
+      id: genver
+      run: |
+        VERSION=$(docker run --rm -v $(pwd):/repo codacy/git-version /bin/git-version --folder=/repo --release-branch=${{ github.ref_name }}-pre --dev-branch=main)
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+        echo "VERSION : $VERSION"
+        RELEASE=$(docker run --rm -v $(pwd):/repo codacy/git-version /bin/git-version --folder=/repo --release-branch=${{ github.ref_name }} --dev-branch=main)
+        echo "release=$RELEASE" >> $GITHUB_OUTPUT
+        echo "RELEASE : $RELEASE"
+        echo "Version :" >> $GITHUB_STEP_SUMMARY
+        echo "$VERSION" >> $GITHUB_STEP_SUMMARY
+        echo "Release :" >> $GITHUB_STEP_SUMMARY
+        echo "$RELEASE" >> $GITHUB_STEP_SUMMARY
+
+  docker:
+    name: Docker image Build
+    runs-on: ubuntu-latest
+    needs:
+      - versionning
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          - { path: images/pdc-update, name: armonik-pdc-update, platforms: "linux/amd64" }
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
```

Comment:
QEMU may not be needed

Reply:
Yes, it is needed to test on ARM

Reply:
My  bad, this is for the docker building. For now, I don't really support ARM images, so it is indeed not needed, but I would need it as soon as we build ARM images.